cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -81,8 +83,7 @@
   private final Map<String, LocalizeResource> resourcesToLocalize;
   private final Supplier<MapReduceTaskContextProvider> taskContextProviderSupplier;
 
-  private InputFormatProvider inputFormatProvider;
-
+  private Map<String, MapperInput> inputs;
```

Comment:
only for the existing API (single input - `setInput`) method, we use the classname, since the user doesn't necessarily specify the input name there.
For the `addInput` methods, we use the user-specified input name as the input name (key to the map).